#! /bin/sh /usr/share/dpatch/dpatch-run
## 99-unnamed.dpatch by  <abi@debian.org>
## Fix delta problem, see: http://www.zabbix.com/forum/showthread.php?t=4315

@DPATCH@
diff -ruN zabbix-1.1.3.orig/src/zabbix_server/functions.c zabbix-1.1.3/src/zabbix_server/functions.c
--- zabbix-1.1.3.orig/src/zabbix_server/functions.c	2006-10-24 14:11:18.237095056 +0600
+++ zabbix-1.1.3/src/zabbix_server/functions.c	2006-10-24 14:13:38.901710784 +0600
@@ -678,7 +678,7 @@
 	value_double	= 0;
 
 	/* Required for nodata() */
-	item->lastclock = now;
+	/* item->lastclock = now; */
 	
 	if(value->type & AR_UINT64)
 	{
@@ -803,6 +803,8 @@
 		DBexecute(sql);
 	}
 
+	item->lastclock = now;
+	
 	return ret;
 }
 
